src/DevTools/createMockNetworkLayer/index.ts

```diff
@@ -53,22 +54,42 @@ export const createMockFetchQuery = ({
     // We pass this field resolver in so that we can control the resolution
     // logic for all data that relay tries to extract from our mock fixtures.
     fieldResolver: ((source, _args, _context, info) => {
-      // source is null for root fields
-      source =
-        source ||
-        (info.operation.operation === "mutation"
-          ? mockMutationResults
-          : mockData)
+      if (responsePathAsArray(info.path).length === 1) {
```

Comment:
Tiny nit: maybe store the result of `responsePathAsArray(info.path)` and re-use it below?